docs(snapshots): Add iOS setup guide (EME-1030) (#17474)
getsentry/sentry-docs: docs/platforms/apple/common/snapshots/index.mdx

---
title: Set Up Snapshots
sidebar_title: Snapshots
sidebar_order: 4950
sidebar_section: features
description: Set up snapshots for iOS apps using Snapshot Previews or your preferred snapshot library.
beta: true
---

<Include name="feature-available-for-user-group-early-adopter" />

Set up [Snapshots](/product/snapshots/) for your iOS app. Generate snapshots locally or in your own CI using your preferred snapshot library, then upload the generated images to Sentry for image diffing, visual review, and GitHub status checks.

## Step 1: Generate Images

There are two paths for generating snapshot images on iOS:

- **From Snapshot Previews (recommended)** — if you don't generate snapshots yet, use [SnapshotPreviews](https://github.com/EmergeTools/SnapshotPreviews), our no code snapshot generation Swift package that turns every `#Preview` macro and `PreviewProvider` in your project into a snapshot.
- **From an existing snapshot tool** — if you already use [swift-snapshot-testing](https://github.com/pointfreeco/swift-snapshot-testing) or another tool, just point `sentry-cli` at the directory it writes images to.

> 

Reviewer: ```suggestion
These methods are not mutually exclusive and you can use the SnapshotPreviews repo in addition to any existing snapshot tests you have.
```

### From Snapshot Previews (Recommended)

> If you don't generate snapshots yet, we recommend using [SnapshotPreviews](https://github.com/EmergeTools/SnapshotPreviews), our no code snapshot generation Swift package that turns every `#Preview` and `PreviewProvider` in your project into a snapshot.

Reviewer: ```suggestion
[SnapshotPreviews](https://github.com/EmergeTools/SnapshotPreviews) automatically turns every `#Preview` and `PreviewProvider` in your project into a snapshot so you don't have to maintain test code. 
```

 xcodebuild test \
  -scheme YourScheme \
  -sdk iphonesimulator \
  -destination 'platform=iOS Simulator,name=iPhone 17 Pro' \
  -only-testing:YourAppTests/YourAppSnapshotTest \
  CODE_SIGNING_ALLOWED=NO

sentry-cli build snapshots --app-id com.example.your-app $TEST_RUNNER_SNAPSHOTS_EXPORT_DIR
```

Replace the path with the directory your library writes to (for swift-snapshot-testing, that's `YourAppTests/__Snapshots__/YourSnapshotTests`). See [Uploading Snapshots](/product/snapshots/uploading-snapshots/) for the full CLI reference and metadata schema.

## Step 3: Integrate Into CI

Once the local upload succeeds, wire the same commands into your CI. See [Uploading Snapshots](/product/snapshots/uploading-snapshots/#upload-with-ci) for the general GitHub Actions structure.

For end-to-end iOS examples, see the workflows in our [Hacker News sample app](https://github.com/EmergeTools/hackernews):

- [SnapshotPreviews workflow](https://github.com/EmergeTools/hackernews/blob/main/.github/workflows/ios_sentry_upload_snapshots.yml) — builds once, fans out across multiple simulators in a matrix, then aggregates all images into a single upload.

> - [swift-snapshot-testing workflow](https://github.com/EmergeTools/hackernews/blob/main/.github/workflows/ios_sentry_upload_swift_snapshots.yml) — records all snapshots in one job and uploads them via the [Sentry Fastlane plugin](https://github.com/getsentry/sentry-fastlane-plugin).

Reviewer: thoughts on doing something like:
```
### Using Fastlane
Fastlane is an alternative option to uploading via the sentry-cli. 
```
...
then just include this content: https://github.com/getsentry/sentry-docs/pull/17375/changes#diff-da4ea75e3c8ea153daadec9c35c38b06c898a38023a312461b0ec7aeec2ff830R34

Author: See commit 121aa38052899b060689afcd567efe1028494ce7

---
title: Set Up Snapshots
sidebar_title: Snapshots
sidebar_order: 4950
sidebar_section: features
description: Set up snapshots for iOS apps using Snapshot Previews or your preferred snapshot library.
beta: true
---

<Include name="feature-available-for-user-group-early-adopter" />

> Set up [Snapshots](/product/snapshots/) for your iOS app. Generate snapshots locally or in your own CI using your preferred snapshot library, then upload the generated images to Sentry for image diffing, visual review, and GitHub status checks.

Reviewer 2: `l`: IIUC this features could also be used for any non-iOS app as it just takes screenshots as input. I believe the wording could be adapted to correctly reflect this to users.
